cs/src/core/FasterLog/FasterLog.cs

```diff
@@ -167,7 +176,8 @@ public FasterLog(FasterLogSettings logSettings)
             commitPolicy.OnAttached(this);
 
             tolerateDeviceFailure = logSettings.TolerateDeviceFailure;
-            if (logSettings.TryRecoverLatest)
+
+            if (syncRecover)
```

Comment:
Need to check recover latest here as well?

If not, maybe we don't need to separate flags for recover latest and recoversync?

Reply:
The private constructor needed to differentiate itself from the public one, so I made it contingent on recoverSync. It does not look at log settings, instead depends on caller.

Reply:
The CreateAsync call invokes this private constructor with this argument false, and takes care of async recovery itself.